### DISCUSSION TITLE
Docker Compose Up Fails to Build Local Site

### DISCUSSION THREAD

**Discussion by Jettngin** [alshedivat/al-folio#2796] [Q&A] [answered]
After being away from building my local site for a weekend I've come back to the 
```
docker compose pull
docker compose up
```
commands throwing the following error
```
Attaching to jekyll-1
jekyll-1  | Gemfile.lock is tracked by git, keeping it intact
jekyll-1  | bundler: failed to load command: jekyll (/usr/local/bundle/bin/jekyll)
jekyll-1  | /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/definition.rb:599:in `materialize': Could not find css_parser-1.19.0, sass-embedded-1.79.3-x86_64-linux-gnu, rexml-3.3.7, activesupport-7.2.1 in locally installed gems (Bundler::GemNotFound)
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/definition.rb:193:in `specs'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/definition.rb:260:in `specs_for'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/runtime.rb:18:in `setup'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler.rb:164:in `setup'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/setup.rb:32:in `block in <top (required)>'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/ui/shell.rb:173:in `with_level'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/ui/shell.rb:119:in `silence'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/setup.rb:32:in `<top (required)>'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli/exec.rb:56:in `require_relative'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli/exec.rb:56:in `kernel_load'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli/exec.rb:23:in `run'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli.rb:455:in `exec'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli.rb:35:in `dispatch'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli.rb:29:in `start'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/exe/bundle:28:in `block in <top (required)>'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/exe/bundle:20:in `<top (required)>'
jekyll-1  |     from /usr/local/bundle/bin/bundle:25:in `load'
jekyll-1  |     from /usr/local/bundle/bin/bundle:25:in `<main>' 
```

I've looked at various possible solutions, but haven't found anything super helpful yet. Any help would be appreciated!

Am running Windows 10

**Comment by george-gca** [comment]
We are working on this.

**Comment by george-gca** [comment, marked as answer]
I believe this is solved now. Try to update your image and try again.

**Reply by Jettngin** [reply]
Works now! Thanks :)

**Reply by Gpayton787** [reply]
What do you mean by update your image? I'm running into the same problem as him.

**Reply by Gpayton787** [reply]
Here is my error by the way: 
```
"Attaching to jekyll-1
jekyll-1  | Entry point script running
jekyll-1  | Gemfile.lock is tracked by git, keeping it intact
jekyll-1  | bundler: failed to load command: jekyll (/usr/local/bundle/bin/jekyll)
jekyll-1  | /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/definition.rb:599:in `materialize': Could not find mini_racer-0.15.0, css_parser-1.19.0, libv8-node-22.7.0.4-aarch64-linux, sass-embedded-1.79.3-aarch64-linux-gnu, rexml-3.3.7, activesupport-7.2.1 in locally installed gems (Bundler::GemNotFound)
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/definition.rb:193:in `specs'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/definition.rb:260:in `specs_for'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/runtime.rb:18:in `setup'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler.rb:164:in `setup'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/setup.rb:32:in `block in <top (required)>'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/ui/shell.rb:173:in `with_level'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/ui/shell.rb:119:in `silence'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/setup.rb:32:in `<top (required)>'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli/exec.rb:56:in `require_relative'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli/exec.rb:56:in `kernel_load'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli/exec.rb:23:in `run'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli.rb:455:in `exec'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli.rb:35:in `dispatch'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/cli.rb:29:in `start'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/exe/bundle:28:in `block in <top (required)>'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
jekyll-1  |     from /usr/local/bundle/gems/bundler-2.5.22/exe/bundle:20:in `<top (required)>'
jekyll-1  |     from /usr/local/bundle/bin/bundle:25:in `load'
jekyll-1  |     from /usr/local/bundle/bin/bundle:25:in `<main>'
"
```

**Reply by george-gca** [reply]
Update your image I mean the docker image.

```bash
docker compose pull
docker compose up
```

**Reply by Gpayton787** [reply]
I tried that but still continue to receive the above error. Will try updating the entire git repo I suppose, that'll take a while though.

**Reply by george-gca** [reply]
Now I see, probably your `docker-compose.yml` is pointing to the wrong docker image version. Try modifying it to this:

https://github.com/alshedivat/al-folio/blob/d24418786940d46983de8f34825d864e1ac6fe6d/docker-compose.yml#L4

**Reply by satwik-kundu** [reply]
Still getting the same error. Tried all of the above steps.

**Reply by george-gca** [reply]
Try to update your code. We removed some dependencies, so at least your `Gemfile`, `Gemfile.lock`, and maybe `_config.yml` should be updated.

**Reply by Jettngin** [reply]
To get mine working I updated my Gemfile.lock, Gemfile, and entry_point.sh files and used the following commands to build the image:
`docker compose pull`
`docker compose up --build`

I think one of the big solves after updating the Gemfiles was updating my entry_point.sh file (though I'm not sure why I had to other than the file was nonexistent in my directory)

**Reply by satwik-kundu** [reply]
Thanks for the help @george-gca @Jettngin! I switched to using Bundler which seemed to work for me :)

**Reply by sarahsmithtripp** [reply]
Hi I just pulled this image new and am now getting issues with the entry_point.sh files. It does not seem to be finding the files locally? 
`Start: Run in container: /bin/sh -c ./bin/entry_point.sh
/bin/sh: 1: ./bin/entry_point.sh: not found`
this is my first time hosting in docker, but I am trying to run via VScode on a Windows 11 machine. I have WSL install on my computer - but can't get the image to host locally. 
Any advice is much appreciated!

**Reply by george-gca** [reply]
Are you using the devcontainer solution?

**Reply by sarahsmithtripp** [reply]
yes - I am using the devcontainer solution. I tried to run it without using the devcontainer, but now I am not sure how to make sure the folder does not run the dev container. Here is the log it spits out 
`Running the postCreateCommand from devcontainer.json...

[24758 ms] Start: Run in container: /bin/sh -c sh /usr/local/post-create.sh
Fetching bundler-2.5.18.gem
Successfully installed bundler-2.5.18
1 gem installed
Fetching gem metadata from https://rubygems.org/........
Fetching rake 13.2.1
Installing rake 13.2.1
Fetching benchmark 0.4.0
Fetching securerandom 0.4.0
Fetching latex-decode 0.4.0
Fetching drb 2.2.1
Fetching logger 1.6.2
Fetching uri 1.0.2
Fetching base64 0.2.0
Fetching bigdecimal 3.1.8
Installing benchmark 0.4.0
Installing securerandom 0.4.0
Fetching fast-stemmer 1.0.2
Fetching matrix 0.4.2
Installing latex-decode 0.4.0
Fetching crass 1.0.6
Installing drb 2.2.1
Fetching cssminify2 2.0.1
Installing logger 1.6.2
Fetching csv 3.3.0
Installing uri 1.0.2
Installing base64 0.2.0
Fetching deep_merge 1.2.2
Fetching sax-machine 1.3.2
Installing bigdecimal 3.1.8 with native extensions
Installing fast-stemmer 1.0.2 with native extensions
Installing matrix 0.4.2
Fetching htmlcompressor 0.4.0
Installing crass 1.0.6
Fetching mini_mime 1.1.5
Installing cssminify2 2.0.1
Installing csv 3.3.0
Installing deep_merge 1.2.2
Installing sax-machine 1.3.2
Installing htmlcompressor 0.4.0
Fetching jekyll-email-protect 1.1.0
Installing mini_mime 1.1.5
Installing jekyll-email-protect 1.1.0
Fetching jekyll-link-attributes 1.0.1
Installing jekyll-link-attributes 1.0.1
Fetching json 2.9.0
Fetching jekyll-regex-replace 1.1.0
Fetching jekyll-twitter-plugin 2.1.0
Installing jekyll-regex-replace 1.1.0
Fetching unicode_utils 1.4.0
Installing json 2.9.0 with native extensions
Fetching namae 1.2.0
Installing namae 1.2.0
Fetching nokogiri 1.17.1 (x86_64-linux)
Fetching bibtex-ruby 6.1.0
Installing unicode_utils 1.4.0
Fetching uglifier 4.2.1
Installing jekyll-twitter-plugin 2.1.0
Fetching css_parser 1.19.1
Fetching citeproc 1.0.10
Installing bibtex-ruby 6.1.0
Installing css_parser 1.19.1
Installing citeproc 1.0.10
Installing uglifier 4.2.1
Fetching csl 1.6.0
Installing csl 1.6.0
Fetching classifier-reborn 2.3.0
Installing classifier-reborn 2.3.0
Fetching citeproc-ruby 1.1.14
Fetching csl-styles 1.0.1.11
Installing citeproc-ruby 1.1.14
Installing nokogiri 1.17.1 (x86_64-linux)
Fetching loofah 2.23.1
Installing loofah 2.23.1
Fetching feedjira 3.2.3
Installing csl-styles 1.0.1.11
Installing feedjira 3.2.3
Fetching json-minify 0.0.3
Installing json-minify 0.0.3
Fetching google-protobuf 4.29.1 (x86_64-linux)
Fetching multi_xml 0.7.1
Installing multi_xml 0.7.1
Installing google-protobuf 4.29.1 (x86_64-linux)
Fetching httparty 0.22.0
Installing httparty 0.22.0
Fetching sass-embedded 1.82.0 (x86_64-linux-gnu)
Installing sass-embedded 1.82.0 (x86_64-linux-gnu)
Fetching jekyll-archives 2.3.0
Fetching jekyll-get-json 1.0.0
Fetching jekyll-imagemagick 1.4.0
Fetching jekyll-jupyter-notebook 0.0.6
Fetching jekyll-minifier 0.1.10
Fetching jekyll-paginate-v2 3.0.0
Fetching jekyll-scholar 7.1.3
Fetching jekyll-tabs 1.2.1
Installing jekyll-archives 2.3.0
Fetching jekyll-toc 0.19.0
Installing jekyll-get-json 1.0.0
Installing jekyll-imagemagick 1.4.0
Installing jekyll-jupyter-notebook 0.0.6
Installing jekyll-minifier 0.1.10
Installing jekyll-paginate-v2 3.0.0
Installing jekyll-scholar 7.1.3
Installing jekyll-tabs 1.2.1
Installing jekyll-toc 0.19.0
Bundle complete! 23 Gemfile dependencies, 89 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Post-install message from httparty:
When you HTTParty, you must party hard!
Running the postAttachCommand from devcontainer.json...

[60636 ms] Start: Run in container: /bin/sh -c ./bin/entry_point.sh
/bin/sh: 1: ./bin/entry_point.sh: not found
[61580 ms] postAttachCommand failed with exit code 127. Skipping any further user-provided commands.
Done. Press any key to close the terminal.`

Thanks for your help on this! I tried to follow the base instructions for running in Dev Containers, and I think I have that working correctly, but this site does not seem to be building!

**Reply by george-gca** [reply]
Tip: when pasting output content it is always useful to paste it between 3 ticks, like

````
```
This way
```
````

So it displays

```
This way
```

---

Now, I just checked your code, and it is from more than 3 years ago. I'suggest you update your code based on the latest version of our repository, since the errors that are appearing are because the needed files are not yet in your repository, they were added later.

We've made a whole lot of changes since then, including some ways to make sure older versions of our repository keep being able to run after we made some changes, like using docker and tagging the versions correctly. But these changes were only made recently, so to benefit from them I'd suggest migrating to the new code.

I recommend re-installing the new version of the theme from scratch and port over your content and changes from the previous version manually. You can copy your code somewhere else, start fresh and check the differences between your repository and ours (don't forget to check the new directory structure in [CUSTOMIZE.md](https://github.com/alshedivat/al-folio/blob/main/CUSTOMIZE.md)). You can use tools like [meld](https://meldmerge.org/) or [winmerge](https://winmerge.org/) to help in this process. Also, if you have any issues, feel free to ask questions, preferably checking if your problem was not already answered and opening new questions in [Q&A](https://github.com/alshedivat/al-folio/discussions/categories/q-a) if they were not.

**Reply by forrestmckee** [reply]
> Hi I just pulled this image new and am now getting issues with the entry_point.sh files. It does not seem to be finding the files locally? `Start: Run in container: /bin/sh -c ./bin/entry_point.sh /bin/sh: 1: ./bin/entry_point.sh: not found` this is my first time hosting in docker, but I am trying to run via VScode on a Windows 11 machine. I have WSL install on my computer - but can't get the image to host locally. Any advice is much appreciated!

I was having a similar issue that ended up being an `LF` vs `CRLF` issue. If you have git bash installed on your system, run `dos2unix ./bin/entry_point.sh`. After that, I ran `docker compose up --build --force-recreate` and now things are running locally in Docker on Windows 11.

**Reply by sarahsmithtripp** [reply]
Hi 
Thank you both so much - I was able to completely rebase the website and host it in a devcontainer on my local! I do have gitbash on my computer and I had to run `dos2unix ./bin/entry_point.sh` to get it to run on my computer. To run in the devcontainer solution I also had to (a) make sure that VScode was running in admin mode (my computer defaults to local) (b) [make sure that the devcontainer is defaulting to execute in WSL](https://stackoverflow.com/questions/75488495/vscode-create-dev-container-docker-not-found-worked-before)

So in the end - the code was old, but it also took a 3-pronged approach to get it to run and now the site is hosting (yay)! Happy holidays ⛄
